src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -82,7 +83,7 @@ private ResponseEntity postEventInternal(final String eventTypeName,
             final JSONArray eventsAsJsonObjects = new JSONArray(eventsAsString);
 
             final int eventCount = eventsAsJsonObjects.length();
-            eventTypeMetrics.reportSizing(eventCount, eventsAsString.length());
+            eventTypeMetrics.reportSizing(eventCount, eventsAsString.getBytes(Charsets.UTF_8).length);
```

Comment:
I agree, actually I was also surprised by this. It's not as trivial a fix as I initially thought however, and I think it would be better to include it in the discussion for ARUHA-473. What choices we make there will influence how we can solve that issue.